SSA/Projects/RISCV64/Semantics.lean

```diff
@@ -0,0 +1,336 @@
+import SSA.Core.MLIRSyntax.EDSL
+set_option maxHeartbeats 1000000000000000000
+/-!
+  ## Dialect semantics
+  This file contains the semantics for each modelled `RISCV-64` dialect instruction
+  as defined by
+  This give us the guarantee that the dialect semantics faithfully implements the intended
+  Risc-V semantics.
+  The semantic defintions do not contain toInt and toNat when possible. For this
+  we rewrote the Sail semantics into pure bit vector only operations including a proof
+  of correctness.
+  This rewrite should allow to gain in automation using the bv_decide.
+-/
+
+namespace RV64Semantics
+
+def ADDIW_pure64 (imm : BitVec 12) (rs1_val : BitVec 64) :  BitVec 64 :=
+     BitVec.signExtend 64 (BitVec.setWidth 32 (BitVec.add (BitVec.signExtend 64 imm) rs1_val))
+
+-- loads immediate into upper 20 bits and then fills the rest up with 0 and returns it as the result
+def UTYPE_pure64_lui (imm : BitVec 20) (pc : BitVec 64) : BitVec 64 :=
+     BitVec.signExtend 64 (imm ++ (0x000 : (BitVec 12)))
```

Comment:
```suggestion
def UTYPE_pure64_lui (imm : BitVec 20) : BitVec 64 :=
     BitVec.signExtend 64 (imm ++ (0x000 : (BitVec 12)))
```

The 2nd argument is unused. Should we drop it? If it must be there, document in the comment why it is there and name it `_` to avoid the warning.